annotation-server/src/gene-phenotypes/gene-phenotypes.service.ts

```diff
@@ -23,62 +25,89 @@ export class GenePhenotypesService {
         @InjectRepository(GenePhenotype)
         private genePhenotypeRepository: Repository<GenePhenotype>,
     ) {
-        this.hashedPhenotypes = new Map<string, Phenotype>();
+        this.hashedGeneSymbols = new Map();
+        this.hashedPhenotypes = new Map();
     }
 
-    private hashedPhenotypes: Map<string, Phenotype>;
-
     async fetchGenePhenotypes(): Promise<void> {
         await this.clearAllData();
 
         this.logger.log('Fetching gene-phenotype combinations from CPIC.');
         const response = this.httpService.get(
             'https://api.cpicpgx.org/v1/diplotype',
-            { params: { select: 'genesymbol, generesult' } },
+            {
+                params: {
+                    select: [
+                        'genesymbol',
+                        'generesult',
+                        'consultationtext',
+                    ].join(','),
+                },
+            },
         );
 
         const diplotypeDtos: DiplotypeDto[] = (await lastValueFrom(response))
             .data;
 
-        await this.geneSymbolRepository.save(
-            await this.getGeneSymbolDtos(diplotypeDtos),
+        await this.genePhenotypeRepository.save(
+            await this.getGenePhenotypes(diplotypeDtos),
```

Comment:
I think it's a bit unclear that `getGenePhenotypes` has a Seiteneffekt of saving `GeneSymbol`s and `Phenotype`s to the database. Maybe consider a more descriptive name that makes this clear? I unfortunately don't have any idea that isn't overly long. What do you think?